modules/core/CameraThinLens.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "CameraPerspective.h"
+
+namespace rt {
+	// ================================ Perspective Thin Lens Camera Class ================================
+	/**
+	 * @brief Perspective Thin Lens Camera class
+	 * @ingroup moduleCamera
+	 */
+	class CCameraThinLens : public CCameraPerspective
+	{
+	public:
+		/**
+		 * @brief Constructor
+		 * @param resolution The image resolution
+		 * @param pos Camera origin (center of projection)
+		 * @param target The target point
+		 * @param up Camera up-vector
+		 * @param angle (Vertical) full opening angle of the viewing frustum (in degrees)
+		 * @param lensRadius The radius of the lense 
+		 * @param focalDistance distance from Camera origin to the focal plane 
+		 * @param nBlades number of aperture blades of the camera
+		 */
+		DllExport CCameraThinLens(Size resolution, const Vec3f& pos, const Vec3f& dir,
```

Comment:
What if we use not direction, but the target point here and calculate the focal distance based on the distance between origin and target points, _i. e._ from the length of (target - pos) vector?